buildSrc/src/main/kotlin/dev/elide/internal/kotlin/plugin/AbstractInternalPluginHost.kt

```diff
@@ -1,83 +0,0 @@
-/*
- * Copyright (c) 2023 Elide Ventures, LLC.
- *
- * Licensed under the MIT license (the "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at
- *
- *   https://opensource.org/license/mit/
- *
- * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
- * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations under the License.
- */
-
-package dev.elide.internal.kotlin.plugin
-
-import dev.elide.internal.ElideInternalPlugin
-import org.gradle.api.Project
-import org.gradle.api.provider.Provider
-import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
-import org.jetbrains.kotlin.gradle.plugin.KotlinCompilerPluginSupportPlugin
-import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
-import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
-
-/** Base host plugin class for Kotlin plugins provided by the Elide build. */
-abstract class AbstractInternalPluginHost<O: ElideInternalPluginsExtension.PluginConfig> protected constructor (
```

Comment:
i think we will still need this infra for plugins...

Reply:
maybe we can keep it in a separate project inside `tools` instead of `buildSrc`?

Reply:
this particular pattern we will need a lot, i think -- anytime we are injecting our own Kotlin Compiler plugin into our own build. i think it's fine in `tools`, but it's an internal-only class

Reply:
actually I think we could move this into the internal build plugin, which is replacing `buildSrc` anyway